src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,55 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<Machine> machines = new ArrayList<>();
+        if (type.equals(Bulldozer.class)) {
+            machines = new BulldozerProducer().get();
+            return machines;
+
```

Comment:
lets create 3 class variables to avoid creating a new instance of producers each time this method is invoked
ps remember about OOP principles - use abstract reference type